macros/models/_sector/alm/arrakis_compatible_trades.sql

```diff
@@ -0,0 +1,273 @@
+                GROUP BY 1,2,4
+            ) AS p
+        ) AS p_adj
+    ) AS p
+    INNER JOIN arrakis_vaults AS a
+        ON a.pool_address = p.pool_address
+    WHERE price IS NOT NULL
+)
+
+, pool_price_usd_ts AS (
+    {{
+        add_pool_price_usd(
+            pool_prices_cte = 'pool_price_ts'
+        )
+    }}
+)
+
+------------------------------------------------------------------------------------------
+-- GET VOLUME PROVIDED BY EACH VAULT -----------------------------------------------------
+------------------------------------------------------------------------------------------
+
+,vault_swaps as (
+    select s.block_time
+        , s.block_number
+        , s.tx_hash
+        , s.evt_index
+        , s.pool_address
+        , lp.vault_address
+        , greatest(least(s.sqrt_price, sqrt(lp.pb)), sqrt(lp.pa)) as sqrt_price
+        , greatest(least(s.prev_sqrt_price, sqrt(lp.pb)), sqrt(lp.pa)) as prev_sqrt_price
+        , lp.liquidity
+    from liquidity_ts as lp
+    inner join (
+        SELECT s.evt_block_time as block_time
+            , s.evt_block_number as block_number
+            , s.evt_tx_hash as tx_hash
+            , s.evt_index
+            , a.pool_address
+            , lag(cast(s.sqrtPriceX96 as double), 1) over (partition by a.pool_address order by s.evt_block_number, s.evt_index) / pow(2,96) as prev_sqrt_price
+            , cast(s.sqrtPriceX96 as double) / pow(2,96) as sqrt_price
+            , abs(cast(s.amount1 as double)) as volume1
+        FROM {{ Pair_evt_Swap }} AS s
+        INNER JOIN arrakis_vaults AS a
+            ON a.creation_time <= s.evt_block_time AND a.pool_address = s.contract_address
+        {% if is_incremental() %}
+        WHERE
+            {{ incremental_predicate('s.evt_block_time') }}
+        {% endif %}
+    ) AS s ON s.block_time = lp.block_time AND s.pool_address = lp.pool_address
+    -- overlapping ranges only
+    where sqrt(lp.pa) <= (case when s.sqrt_price > s.prev_sqrt_price then s.sqrt_price else s.prev_sqrt_price end)
+        and (case when s.prev_sqrt_price < s.sqrt_price then s.prev_sqrt_price else s.sqrt_price end) <= sqrt(lp.pb)
+) --select * from vault_swaps
+
+select distinct '{{ blockchain }}' AS blockchain
+    , '{{ project }}' AS project
+    , '{{ version }}' AS version
+    , v.block_time
+    , date_trunc('MONTH',v.block_time) AS block_month
+    , v.block_number
+    , v.tx_hash
+    , v.evt_index
+    , a.pool_address
+    , a.vault_address
+    , a.token0_address
+    , a.token1_address
+    , v.volume1 AS volume1_raw
+    , v.volume1 / p.price AS volume0_raw
+    , v.volume1 / power(10, i.decimals1) AS volume1
+    , v.volume1 / (p.price * power(10, i.decimals0)) AS volume0
+    , p.price
+    , p.price_usd
+    , v.volume1 / power(10, i.decimals1) * p.price_usd AS volume_usd
+    , v.volume1 / (p.price * power(10, i.decimals0)) * p.price_usd AS volume_usd_2
```

Comment:
Hi @0xrusowsky, reference `i` is unclear.
I am going to comment these lines first.